File: backend/src/main/java/com/davnig/invoicemanagementapi/model/entity/FlatRateScheme.java
package com.davnig.invoicemanagementapi.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "t_flat_rate_scheme")
@Getter
@Setter
public class FlatRateScheme {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "flat-rate-scheme-id-gen")
    @SequenceGenerator(name = "flat-rate-scheme-id-gen", sequenceName = "t_flat_rate_scheme_id_seq", allocationSize = 1)
    private int id;
    private String name;
    private float rate;

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/entity/FlatRateScheme.kt
package com.davnig.invoicemanagementapi.model.entity

import jakarta.persistence.*

@Entity
@Table(name = "t_flat_rate_scheme")
class FlatRateScheme {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "flat-rate-scheme-id-gen")
    @SequenceGenerator(name = "flat-rate-scheme-id-gen", sequenceName = "t_flat_rate_scheme_id_seq", allocationSize = 1)
    var id = 0
    var name: String? = null
    var rate = 0f

}